server.js

```diff
@@ -5,37 +5,65 @@ const getNews = require('./newsAPI');
 
 //functions to talk to DB
 
-let createUser = (username, password, leaning, email) => {
+let createUserDb = (username, password, leaning, email) => {
   return db.query(`INSERT INTO users
     (username, password, leaning, email)
     VALUES('${username}', '${password}', '${leaning}', '${email}');`);
 };
 
-let rateArticle = (userid, articleid, rating) => {
+let getUserDb = (id) => {
+  return db.query(`SELECT * from users where userid = ${id}`);
+}
+
+let rateArticleDb = (userid, articleid, rating) => {
   return db.query(`INSERT INTO ratings(userid, articleid, ${rating}) VALUES('${userid}', '${articleid}', 1);`);
 };
 
-let addArticle = (url, author, description, publishedAt, source, urlToImage, topic) => {
+let addArticleDb = (url, author, description, publishedAt, source, urlToImage, topic) => {
   return db.query(`INSERT INTO articles(topic, url, author, description, publishedAt, source, urlToImage)
   VALUES('${topic}', '${url}', '${author}', '${description}', '${publishedAt}', '${source}', '${urlToImage}');`);
 };
 
-let matches = function(request, method, path) {
+//helper functions
+
+let getSuffix = (fullUrl, prefix) => fullUrl.slice(prefix.length);
+
+//handlers
+
+let getUser = (request, response) => {
+  let id = getSuffix(request.url, '/users/');
+  getUserDb(id).then((data) => response.end(JSON.stringify(data)));
+}
+
+//Routes and server
+
+let matches = (request, method, path) => {
   return request.method === method &&
          path.exec(request.url);
 };
 
+let notFound = (request, response) => {
+  response.statusCode = 404;
+  response.end('404, nothing here!');
+};
+
 let routes = [
   { method: 'DELETE', path: /^\/users\/([0-9]+)$/, handler: deleteUser },
   { method: 'GET', path: /^\/users\/([0-9]+)$/, handler: getUser },
   { method: 'PUT', path: /^\/users\/([0-9]+)$/, handler: putUser },
   { method: 'GET', path: /^\/users\/?$/, handler: getUsers },
   { method: 'POST', path: /^\/users\/?$/, handler: postUser },
+  { method: 'GET', path: /^\/signin\/?$/, handler: signIn },
```

Comment:
I realized last night that a 'GET' method cannot have a body attached to it so can you change it to a 'POST' method. Other than that - things look good. Once you merge this - I'll merge my code that validates a userID and password against the DB.

Reply:
updated.  were you able to use the method with your local db?

Reply:
yeah, I wrote a test one last night and passed some test data to it and then wrote a validation function to call the db and compare the two.